src/core_plugins/timelion/public/directives/timelion_expression_input.js

```diff
@@ -150,16 +150,26 @@ app.directive('timelionExpressionInput', function ($document, $http, $interval,
       }
 
       async function getSuggestions() {
+        scope.isLoadingSuggestions = true;
+        scope.suggestions.reset();
+        scope.suggestions.show();
+
         const suggestions = await suggest(
-          scope.sheet,
+          scope.sheet,  // sheet is the timelion expression
```

Comment:
I just had a thought... I wonder if you could avoid saving the expression on the returned expressions if you just saved a local variable here.  
```
const expression = scope.sheet; // maybe need to make a copy??
const suggestions = await suggest(expression...);
if (scope.sheet !== expression) {
  // Expression changed since requesting, no longer valid
  return;
}
```

Could then lessen the footprint of this PR even more.